frontend/src/app/components/search-form/search-form.component.ts

```diff
@@ -96,21 +96,26 @@ export class SearchFormComponent implements OnInit {
     if (searchText) {
       this.isSearching = true;
       if (this.regexAddress.test(searchText)) {
-        this.navigate('/address/', searchText);
-      } else if (this.regexBlockhash.test(searchText) || this.regexBlockheight.test(searchText)) {
+        const matches = this.regexAddress.exec(searchText);
+        this.navigate('/address/', matches[0]);
+      } else if (this.regexBlockhash.test(searchText)) {
+        const matches = this.regexBlockhash.exec(searchText);
+        this.navigate('/block/', matches[0]);
+      } else if (this.regexBlockheight.test(searchText)) {
         this.navigate('/block/', searchText);
       } else if (this.regexTransaction.test(searchText)) {
+        const matches = this.regexTransaction.exec(searchText);
         if (this.network === 'liquid') {
-          if (this.assets[searchText]) {
-            this.navigate('/asset/', searchText);
+          if (this.assets[matches[0]]) {
```

Comment:
@softsimon I don't think we will ever reach this since we are comparing assets with a tx regex?